lib/ansible/plugins/strategy/__init__.py

```diff
@@ -928,6 +928,8 @@ def _execute_meta(self, task, play_context, iterator, target_host):
         meta_action = task.args.get('_raw_params')
 
         def _evaluate_conditional(h):
+            if task.when == []:
```

Comment:
We can have `task.when == False` which would mean the opposite of 'no conditions' , which is what `[]` represents

Reply:
The idea here was to short-circuit only the default value and everything else should go through the normal conditional evaluation. `when: []` and `when: ""` are True, `when: 0` and `when: False` are False. I would be worried about this diverging with normal conditional evaluation.

Reply:
Since `when` is a list type, anything explicitly provided is not boolean false iirc.  They turn into `[[]]`, `[""]`, `[0]`, `[False]`. But regardless, `== []` or just a boolean evaluation doesn't really impact the outcome.

Reply:
Ok, fixed.